src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -326,23 +280,26 @@ private void WatchForFileSystemEventsThreadStart()
                         // Note: the order here matters due to coalescing since two events (such as a mod followed by a delete) could be coalesced into one notification
                         //       So look for deletes first since those will mean the listening app can't do anything with the item since it doesn't exist (such as stat 
                         //       the file to determine what exactly changed, in this example).
-                        if ((eventFlags[i] & Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRemoved) == Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRemoved)
-                        {
-                            NotifyFileSystemEventArgs(WatcherChangeTypes.Deleted, relativePath);
-                        }
-                        else if ((eventFlags[i] & Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated) == Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated)
+                        if ((eventFlags[i] & Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated) == Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated)
```

Comment:
The length of these fully-qualified consts makes it a bit difficult to read, and the duplication of the name can lead to accidental typos; did you consider adding a helper somewhere like:
```C#
private static bool IsSet(int value, int flags)
{
    return (value & flags) != flags;
}
```
and then using it like:
```C#
if (IsSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemCreated) { ... }
```